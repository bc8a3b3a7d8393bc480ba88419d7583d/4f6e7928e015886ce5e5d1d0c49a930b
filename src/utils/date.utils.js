import moment from 'moment';

const supportedDateFormats = {
  ddddDDMMMYYYY: 'dddd, DD MMM YYYY',
};

const format = (date, dateFormat) => moment(date).format(dateFormat);

export default {
  format,
  supportedDateFormats,
};
