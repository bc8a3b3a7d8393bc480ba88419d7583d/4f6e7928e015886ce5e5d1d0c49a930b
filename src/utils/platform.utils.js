import { Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import { equals } from 'ramda';

const isDEV = () => equals(process.env.NODE_ENV, 'development');
const isIOS = () => equals(Platform.OS, 'ios');
const isAndroid = () => equals(Platform.OS, 'android');
const iPhoneX = 'iPhone X';
const isIPhoneX = () => isIOS() && equals(DeviceInfo.getModel(), iPhoneX);

export { isDEV, isIOS, isAndroid, isIPhoneX, iPhoneX };
