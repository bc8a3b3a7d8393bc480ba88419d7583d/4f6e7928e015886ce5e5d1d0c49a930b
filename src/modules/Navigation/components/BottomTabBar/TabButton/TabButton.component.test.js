import React from 'react';
import { shallow } from 'enzyme';
import { testSnapshots } from '../../../../../utils/test.utils';

import Component from './TabButton.component';

describe('Navigation/Component - TabButton', () => {
  const componentProps = {
    title: 'Tab Button',
    route: 'Home',
    onPress: jest.fn(),
  };

  const render = props => shallow(<Component {...props} />);

  testSnapshots(Component, [
    {
      description: 'should render default state',
      props: componentProps,
    },
    {
      description: 'should render with selected state',
      props: {
        ...componentProps,
        selected: true,
      },
    },
  ]);

  test('should call onPress prop when the user taps on tab bar item', () => {
    const component = render(componentProps);
    component.find('TouchableWithoutFeedback').simulate('press');

    expect(componentProps.onPress).toBeCalledWith(componentProps.route);
  });
});
