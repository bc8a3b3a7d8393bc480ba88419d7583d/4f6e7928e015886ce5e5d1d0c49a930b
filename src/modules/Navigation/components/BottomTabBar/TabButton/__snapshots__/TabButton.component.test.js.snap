// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation/Component - TabButton snapshots should render default state 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID=""
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/navigation/home.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#8f8f8f",
            "fontFamily": "Oswald-Regular",
            "fontSize": 12,
          },
          false,
        ]
      }
    >
      Tab Button
    </Text>
  </Component>
</TouchableWithoutFeedback>
`;

exports[`Navigation/Component - TabButton snapshots should render with selected state 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID=""
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/navigation/home_fill.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#8f8f8f",
            "fontFamily": "Oswald-Regular",
            "fontSize": 12,
          },
          Object {
            "color": "#0d7fcc",
          },
        ]
      }
    >
      Tab Button
    </Text>
  </Component>
</TouchableWithoutFeedback>
`;
